### PR TITLE
Improve documentation for the operator in OpenShift

### PR DESCRIPTION
**Pull request by arapulido**
### What does this PR do?

It adds new documentation to deploy the operator using OLM and the OperatorHub specific to OpenShift

### CODE REVIEW

File: examples/datadogagent/datadog-agent-openshift.yaml [hunk 1]
```diff
@@ -2,26 +2,33 @@ apiVersion: datadoghq.com/v1alpha1
...
   clusterAgent:
-    config:
-      externalMetrics:
-        enabled: true
-        port: 5010
-      clusterChecksRunnerEnabled: true
-    replicas: 2
-  clusterChecksRunner:
-    enabled: true
+    replicas: 1
```
Comment (clamoriniere) [inline examples/datadogagent/datadog-agent-openshift.yaml:34]: this part is not mandatory. 
the clusterAgent is enable by default with 1 replica

Comment (arapulido) [reply examples/datadogagent/datadog-agent-openshift.yaml]: removed

File: docs/install-openshift.md [hunk 2]
```diff
@@ -0,0 +1,76 @@
+# Installation
```
Comment (alai97) [inline docs/install-openshift.md:1]:
```suggestion
## Overview
```

File: docs/install-openshift.md [hunk 3]
```diff
@@ -0,0 +1,76 @@
...
+
+The Datadog Operator is [certified by RedHat's Marketplace][1].
+
+## Deploy the Datadog Operator
```
Comment (alai97) [inline docs/install-openshift.md:5]:
```suggestion
## Deploy the Datadog Operator in an OpenShift cluster
```

File: docs/install-openshift.md [hunk 4]
```diff
@@ -0,0 +1,76 @@
...
+
+To deploy the Datadog Operator in an OpenShift cluster we recommend using the [Operator Lifecycle Manager][2], deploying the operator directly from the OperatorHub in your OpenShift cluster Web Console.
```
Comment (alai97) [inline docs/install-openshift.md:7]:
```suggestion
Use the [Operator Lifecycle Manager][2] to deploy the Datadog Operator from OperatorHub in your OpenShift Cluster web console.
```

File: docs/install-openshift.md [hunk 5]
```diff
@@ -0,0 +1,76 @@
...
+
+1. Create a new `datadog` project in your OpenShift cluster:
```
Comment (alai97) [inline docs/install-openshift.md:9]:
```suggestion
1. Create a `datadog` project in your OpenShift cluster:
```

File: docs/install-openshift.md [hunk 6]
```diff
@@ -0,0 +1,76 @@
...
+
+   ```shell
+   oc new-project datadog
+   ```
+1. In OperatorHub, in the OpenShift Web Console, search of the Datadog Operator and click Install:
```
Comment (alai97) [inline docs/install-openshift.md:14]: Are you referencing to two locations or one here?

```suggestion

2. In OperatorHub or the OpenShift Web Console, search for the Datadog Operator and click **Install**.
```

File: docs/install-openshift.md [hunk 7]
```diff
@@ -0,0 +1,76 @@
...
+
+![Datadog Operator in the OperatorHub](assets/operatorhub.png)
+
+1. In the Install options, select “datadog” as the Installed Namespace:
```
Comment (alai97) [inline docs/install-openshift.md:18]:
```suggestion
3. Select `datadog` as the **Installed Namespace**:
```

File: docs/install-openshift.md [hunk 8]
```diff
@@ -0,0 +1,76 @@
...
+
+![Deploy the operator in the datadog namespace](assets/datadognamespace.png)
+
+## Deploy the Datadog Agents with the operator
+
+After deploying the Datadog Operator, create the `DatadogAgent` resource that triggers the Datadog Agent's deployment in your OpenShift cluster. By creating this resource, the Agent will be deployed as a `DaemonSet` on every `Node` of your cluster.
```
Comment (alai97) [inline docs/install-openshift.md:24]:
```suggestion
After deploying the Datadog Operator, create a `DatadogAgent` resource that triggers the Datadog Agent's deployment in your OpenShift cluster. With this resource, the Agent deploys as a `DaemonSet` on every `Node` of your cluster.
```

File: docs/install-openshift.md [hunk 9]
```diff
@@ -0,0 +1,76 @@
...
+
+1. Create a Kubernetes secret with your API and APP keys
```
Comment (alai97) [inline docs/install-openshift.md:26]:
```suggestion
1. Create a Kubernetes secret with your API and APP keys:
```

File: docs/install-openshift.md [hunk 10]
```diff
@@ -0,0 +1,76 @@
...
+
+   ```shell
+   kubectl create secret generic datadog-secret -n datadog --from-literal api-key=<DATADOG_API_KEY> --from-literal app-key=<DATADOG_APP_KEY>
+   ```
+   Replace `<DATADOG_API_KEY>` and `<DATADOG_APP_KEY>` with your [Datadog API and application keys][3]
+
+1. Create a file with the spec of your DatadogAgent deployment configuration. The simplest configuration is:
```
Comment (alai97) [inline docs/install-openshift.md:33]:
```suggestion
2. Create a file with the spec of your `DatadogAgent` deployment configuration. 

The following contains the simplest configuration:
```

File: docs/install-openshift.md [hunk 11]
```diff
@@ -0,0 +1,76 @@
...
+              type: spc_t
+              user: system_u
+          criSocket:
+            criSocketPath: /var/run/crio/crio.sock
+          env:
+          - name: DD_KUBELET_TLS_VERIFY
+            value: "false"
+   ```
+
+1. Deploy the Datadog agent with the above configuration file:
```
Comment (alai97) [inline docs/install-openshift.md:67]:
```suggestion
3. Deploy the Datadog Agent with the configuration file above:
```

File: docs/install-openshift.md [hunk 12]
```diff
@@ -0,0 +1,76 @@
...
+   ```shell
+   oc apply -f path/to/your/datadog-agent.yaml
+   ```
+
+The Datadog Agent will run with the service account `datadog-agent-scc` created by the Datadog Operator and connected to the right SCCs.
```
Comment (alai97) [inline docs/install-openshift.md:72]:
```suggestion
The Datadog Agent runs with the `datadog-agent-scc` service account created by the Datadog Operator and is connected to the right SCCs.
```

**Review by alai97 — CHANGES_REQUESTED** [review]
Doc review!

**Review by vboulineau — APPROVED** [review]

**Review by alai97 — APPROVED** [review]

**Comment by arapulido** [comment]
@clamoriniere any thing else before we merge? Thanks!

**Review by clamoriniere — APPROVED** [review]

**Pull request merged**